awkernel_async_lib/src/pubsub.rs

```diff
@@ -176,6 +166,30 @@ impl<T: Clone + Send> Subscriber<T> {
         receiver.await
     }
 
+    pub(super) fn recv_or_register_waker(
```

Comment:
Could you write the documentation comments that describes the lost wake situation?